spec/core/ics-033-multi-hop/README.md

```diff
@@ -0,0 +1,463 @@
+
+For each multi-hop channel (detailed proof logic below):
+
+1. Scan source chain for IBC messages to relay.
+2. Read the connectionHops field in from the scanned message to determine the channel path.
+3. Lookup connection endpoints via chain registry configuration and ensure each connection in the channel path is updated to include the key/value to be proven on the source chain.
+4. Query for proof of connection, and consensus state for each intermediate connection in the channel path to the destination connection.
+5. Query proof of packet commitment or handshake message commitment on source chain.
+6. Submit proofs and data to RPC endpoint on destination chain.
+
+Relayers are connection topology aware with configurations sourced from the [chain registry](https://github.com/cosmos/chain-registry).
+
+### Proof Generation & Verification
+
+Graphical depiction of proof generation.
+
+![proof_generation.png](proof_generation.png)
+
+Relayer multi-hop proof queries.
+
+![proof_query.png](proof_query.png)
+
+Multi-hop proof verfication logic.
+
+![proof_verification.png](proof_verification.png)
+
+Pseudocode proof generation for a channel between `N` chains `C[0] --> C[i] --> C[N]`
+
+```go
+
+// ProofData is a generic proof struct.
+type ProofData struct {
+    Key   *MerklePath
+    Value []]byte
+    Proof []byte
+}
+
+// MultihopProof defines set of proofs to verify a multihop message.
+// Consensus and Connection proofs are ordered from source to destination but not including the
+// chain[N-1] consensus state on chainN (dest) since it is already known on the destination chain.
+type MultihopProof struct {
+    KeyProofIndex uint32           // the index of the consensus state to prove the key/value on, normally 0
+    KeyProof *ProofData            // the key/value proof on the source chain (consensusState[0])
+    ConsensusProofs []*ProofData   // array of consensus proofs starting with proof of consensusState[0] state root on chain1 (consensusState[1])
+    ConnectionProofs []*ProofData  // array of connection proofs starting with proof of conn[1,0] on chain1 (consensusState[1])
+}
+
+// GenerateMultihopProof generates proof of a key/value at the proofHeight on source chain (chain0). 
+func GenerateMultihopProof(chains []*Chain, key string, value []byte, proofHeight exported.Height) *MultihopProof {
+
+    abortTransactionUnless(len(chains) > 2)
+
+    var multihopProof MultihopProof
+    chain0 := chains[0] // source chain
+    chain1 := chains[1] // first hop chain
+ 
+    height01 := chain1.GetClientStateHeight(chain0) // height of chain0's client state on chain1
+    abortTransactionUnless(height01 >= proofHeight) // ensure that chain0's client state is update to date
```

Comment:
Hmm, looking at this pseudo code I think it needs some clarification.

Really what this code is trying to do is ensure that each client in the multi-hop channel path is up to date with respect to the prior chain. The key to be proven on the source chain (chain0) is committed at some height so can be proven in chain0's consensusState on chain1 at that specific height. This is why the check is ">=". Does that make sense or maybe I missed something?

Its a bit confusing thought  since this is not really the `proofHeight` which has a different meaning. I think maybe it could be called `keyHeight`? Each subsequent client state needs to be up to date to ensure the prior chain's consensusState is included and that the consensusState at `keyHeight` on chain1 includes the key/value on the source chain.

The proofHeight during verification is used to indicate the height of the N-1'th consensus/connection on the N'th (receiving) chain.